app/code/Magento/InventoryInStorePickup/Model/IsOrderReadyForPickup.php

```diff
@@ -28,29 +31,70 @@ class IsOrderReadyForPickup implements IsOrderReadyForPickupInterface
      */
     private $orderRepository;
 
+    /**
+     * @var OperationResultInterfaceFactory
+     */
+    private $operationResultFactory;
+
     /**
      * @param IsFulfillable $isFulfillable
      * @param OrderRepositoryInterface $orderRepository
+     * @param OperationResultInterfaceFactory $operationResultFactory
      */
     public function __construct(
         IsFulfillable $isFulfillable,
-        OrderRepositoryInterface $orderRepository
+        OrderRepositoryInterface $orderRepository,
+        OperationResultInterfaceFactory $operationResultFactory
     ) {
         $this->isFulfillable = $isFulfillable;
         $this->orderRepository = $orderRepository;
+        $this->operationResultFactory = $operationResultFactory;
     }
 
     /**
      * Check if order can be shipped and the pickup location has enough QTY.
      *
-     * @param int $orderId
-     * @return bool
+     * @param int[] $orderIds
+     * @return OperationResultInterface
      */
-    public function execute(int $orderId): bool
+    public function execute(array $orderIds): OperationResultInterface
```

Comment:
Should service be renamed since it's now working with multiple entities? Is there a naming convention for this case in place?